thentos-core/src/Thentos/Backend/Core.hs

```diff
@@ -80,13 +81,15 @@ newtype ErrorMessage = ErrorMessage { fromErrorMessage :: ST }
 instance ToJSON ErrorMessage where
     toJSON (ErrorMessage msg) = object ["status" .= String "error", "error" .= msg]
 
+-- |Type of functions that transform a base error and an error message into a ServantErr.
+type MkServantErrFun = ServantErr -> ST -> ServantErr
```

Comment:
I prefer not using type synonyms here, since it's much less obvious what type `mkServantErr` actually expects by looking just at its signature.